backup/data.go

```diff
@@ -154,7 +154,7 @@ func backupDataForAllTables(tables []Table) []map[uint32]int64 {
 				// Afterwards, we break early and terminate the worker since its transaction is now in an
 				// aborted state. We do not need to do this with the main worker thread because it has
 				// already acquired AccessShareLocks on all tables before the metadata dumping part.
-				if whichConn != 0 {
+				if whichConn != 0 && table.ForeignDef == (ForeignTableDefinition{}) {
```

Comment:
Could you explain why `&&` is used here, not `||`? Isn't this branch also used for some other special kinds of tables?

Reply:
`table.ForeignDef == (ForeignTableDefinition{})` mains, that the Table struct doesn't contain **filled** foreign table definition, thus it can be of any table type except a foreign one.